cli/command/environment.go

```diff
@@ -217,42 +231,52 @@ func (e *EnvironmentCommand) update(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) link(c *cli.Context) error {
-	args, err := extractArgs(c.Args(), "SOURCE_ENVIRONMENT_NAME", "DESTINATION_ENVIRONMENT_NAME")
-	if err != nil {
-		return err
-	}
+	generateAddLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	sourceEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
-	}
+		req := models.UpdateEnvironmentRequest{EnvironmentID: sourceEnvironmentID}
+		appendedLinks := append(env.Links, destEnvironmentID)
+		req.Links = &appendedLinks
 
-	destEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["DESTINATION_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
+		return req, nil
 	}
 
-	if sourceEnvironmentID == destEnvironmentID {
-		return fmt.Errorf("Cannot link an environment to itself")
-	}
+	return e.updateEnvironmentLinksHelper(c, generateAddLinkRequest)
+}
 
-	req := models.CreateEnvironmentLinkRequest{
-		SourceEnvironmentID: sourceEnvironmentID,
-		DestEnvironmentID:   destEnvironmentID,
-	}
+func (e *EnvironmentCommand) unlink(c *cli.Context) error {
+	generateRemoveLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	jobID, err := e.client.CreateLink(req)
-	if err != nil {
-		return err
+		destEnvIDIndex := -1
+		for i, link := range env.Links {
+			if link == destEnvironmentID {
+				destEnvIDIndex = i
+				break
+			}
+		}
+
+		if destEnvIDIndex >= 0 {
+			env.Links = append(env.Links[:destEnvIDIndex], env.Links[destEnvIDIndex+1:]...)
+		}
+
+		req := models.UpdateEnvironmentRequest{EnvironmentID: sourceEnvironmentID}
```

Comment:
`EnvironmentID` field should not exist

Reply:
? can you elaborate - not having environment id here would require changing the client interface for all the entities.